### ISSUE TITLE
[Documentation] Outputpanel: wrong description

### ISSUE THREAD

**Issue by Rapster** [primefaces/primefaces#2674]
I think the placeholder of outputpanel is kinda wrong, I quote:

> When a JSF component is not rendered, no markup is rendered so for components with conditional
> rendering, regular update mechanism may not work since the markup to update on page does not
> exist. OutputPanel is useful in this case to be used as a placeholder.
> Suppose the rendered condition on bean is false when page if loaded initially and search method on
> bean sets the condition to be true meaning datatable will be rendered after a page submit. The
> problem is although partial output is generated, the markup on page cannot be updated since it
> doesn't exist.
>
> Original issue: https://forum.primefaces.org/viewtopic.php?f=3&t=52274

I don't think that's the purpose of this component since you can do it in native JSF, see https://stackoverflow.com/questions/9010734/why-do-i-need-to-nest-a-component-with-rendered-some-in-another-component-w  

To me, the real benefit of this component is the deferred part

**Comment by tandraschko** [comment]
hmmm? the description sounds correct for me - also if the real benefit might be deffered loading and autoUpdate etc.

**Comment by Rapster** [comment]
By "wrong" I mean, this is not the main benefit of this component, like I said you can do exactly the same in native JSF and as simple. The description mislead developers of my company tending to use this component while it's not necessary...

**Comment by tandraschko** [comment]
Should we change the descrption cagatay? IMO it's not nessecary but we should think about it.

**Comment by Rapster** [comment]
@tandraschko @cagataycivici In case (and IMO it's necessary) you decide to remove this description, I think `deferred` should be set to true by default

**Comment by rapster** [comment]
One more thing, if the placeholder is correct, this code should work (but it's not unless I miss something):
```xml
<p:outputPanel id="panel" rendered="#{autoCompleteView.truc}">
    <h3>Loaded after page load.</h3>
    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
</p:outputPanel>

<p:commandButton value="Update" update="panel">
    <f:setPropertyActionListener target="#{autoCompleteView.truc}" value="true" />
</p:commandButton>
```

**Comment by Rapster** [comment]
That description is still there, which again I think is misleading the main benefit of this component. Renaming that component into `<p:lazyPanel />` would also help IMO

**Comment by Rapster** [comment]
I'd like to reopen that ticket and have @melloware @cnsgithub opinion. @tandraschko I'm not sure what you think, but I'd like you to reconsider it :smile: . IMO, and I'll repeat myself:

- Description does not describe the real benefit of this component, instead it says something that you can do natively in JSF, while it does not mention clearly the main feature of this component. This is why I said it's a wrong description

- The name `outputpanel` does not help, a panel in the general term is an output component, so what outputpanel means?

- Being said,  a lot of developers (PF team first, see showcase) sadly, will use this component while it's not necessary.

Basically, what I want to do:
- Rename into `p:lazyPanel` (the name says already a lot about what the component does)
- Having `mode` attribute (either `load` or `visible`, it could also be interesting to set a custom callback)
- Having `disabled` attribute (in case it has to be dynamically disabled for some reasons)

Thanks for your feedbacks

**Comment by tandraschko** [comment]
thats up to primetek
WDYT @mertsincan?

**Comment by melloware** [comment]
@Rapster I know I have tons of code using p:outputPanel because in JSF 2.0 and 2.1 we would use it as a wrapper component for the rendered="" flag.  Because before JSF 2.2 we couldn't do `jsf:rendered="#{value}"`

And those were how all the early examples from PF were back in 2011 so I think that is why they named it OutputPanel because it was just a `<div>` allowing you to control whether the output was shown or not with the Rendered flag.  In the years since it seems to have grown in functionality with deferred functionality.

Renaming it to p:lazyPanel i am not sure gives the benefit vs the last 9-10 years of code people using it as p:outputPanel.  Just my two cents. 

It feels like a very expensive change.  I can see maybe leave p:outputPanel tag in the taglib to maintain backward compatibility?

**Comment by Rapster** [comment]
> @Rapster I know I have tons of code using p:outputPanel because in JSF 2.0 and 2.1 we would use it as a wrapper component for the rendered="" flag. Because before JSF 2.2 we couldn't do jsf:rendered="#{value}"
> 
> And those were how all the early examples from PF were back in 2011 so I think that is why they named it OutputPanel because it was just a <div> allowing you to control whether the output was shown or not with the Rendered flag. In the years since it seems to have grown in functionality with deferred functionality.

I don't get it, why not simply use `h:panelGroup`?

**Comment by melloware** [comment]
There is no reason we could not use h:PanelGroup other than like I said all the PrimeFaces examples used p:outputPanel and when we started in 2011 we were given the directive to use PF component anytime one existed and only use h: components when there was no PF equivalent.

**Comment by Rapster** [comment]
> And those were how all the early examples from PF were back in 2011 so I think that is why they named it OutputPanel because it was just a <div> allowing you to control whether the output was shown or not with the Rendered flag.

I think h:panelGroup was available at that time, `rendered` and `layout` attributes too

> There is no reason we could not use h:PanelGroup other than like I said all the PrimeFaces examples used p:outputPanel 

Yeah I understand, I kinda blame the showcase for that, it misguides a lot of developers like myself. Same scenario goes for using `actionListener` instead of `listener` (something I recently changed)

> when we started in 2011 we were given the directive to use PF component anytime one existed and only use h: components when there was no PF equivalent.

It makes sense for visual components, but outputPanel is not really one of them if you use it with deferred disabled.

**Comment by tandraschko** [comment]
+0 from my side
the benefits are only a bit better name
we could also just do a better showcase description

also note:
not everyone uses JSF 2.2+, so jsf:rendered is not available

**Comment by tandraschko** [comment]
(also we can always improve docu + showcase but we can't rename such old components without talking with primetek)

**Comment by Rapster** [comment]
Yeah well the name would make a lot of difference imo, but as a starter definitely change the doc description... And I don't think passthrough attributes would solve anything in this scenario

**Comment by melloware** [comment]
> Yeah well the name would make a lot of difference imo, but as a starter definitely change the doc description... And I don't think passthrough attributes would solve anything in this scenario

No what I think he is saying is in my case where i was using p:outputPanel I can just do

```xml
<div jsf:rendered="#{value}">Output Here</div>
```

is only available in JSF 2.2+

**Comment by tandraschko** [comment]
Changed the docs a bit now.
